### PR TITLE
MEN-6861: Add `mender-snapshot` to the image and enable and fix the tests

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline --pr mender-snapshot/15

**Comment by lluiscampos** [comment]
> 👨 🏗️ 👉 [Pipeline 1103117685](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1103117685)

The snapshot tests passed in this pipeline. Many other tests failed due to this branch not being up to date to the feature branch, specifically missing the `mender-flash` 1.0.0 recipe which made the build use an old version.

The PR can be reviewed anyway and I will trigger a new build if changes are required.

File: meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc [hunk 1]
```diff
@@ -19,7 +19,7 @@ PACKAGES =+ "mender-auth mender-update mender-config mender-auth-dev"
 RDEPENDS:mender-auth += "mender-config"
 
 RDEPENDS:mender-update += "mender-auth mender-config"
-RRECOMMENDS:mender-update += "mender-flash"
+RRECOMMENDS:mender-update += "mender-flash mender-snapshot"
```
Comment (lluiscampos) [inline meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc:22]: Discussed f2f,

Move the recommendation to `meta-mender-demo` layer :+1:

**Comment by lluiscampos** [comment]
New build @ [Pipeline 1103658776](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1103658776)

**Review by oleorhagen — APPROVED** [review]
Good thing you had to wait a little for the tests :)

**Review by kacf — APPROVED** [review]

**Pull request merged**